docs/databases/swift-kuery.html

```diff
@@ -252,237 +229,150 @@ <h2 class="heading-2"><span class="blue-text">Step 5 (Optional):</span> Test sav
-          }
-          return callback("Error executing query: \(error)")
-        }
-      }
+      <pre><code class="language-swift">connection.execute(query: selectQuery) { selectResult in
+    guard let resultSet = selectResult.asResultSet else {
+        Log.error("Error connecting: \(selectResult.asError?.localizedDescription ?? "Unknown Error")")
+        return completion(nil, .internalServerError)
     }
-  }</code></pre>
+    // Iterate through result set here
+}</code></pre>
       <p>Here we check to see if the `selectResult` value is a valid Result set otherwise we log and error and return.</p>
-      <p>Next we need to create a variable that will store the value we respond to the incoming GET request with:</p>
-      <pre><code class="language-swift">func getTitles(_ callback: @escaping (String) -> Void ) -> Void {
-    pool.getConnection() { connection, error in
-      guard let connection = connection else {
-        guard let error = error else {
-          Log.error("Error connecting: Unknown Error")
-          return
-        }
-        Log.error("Error connecting: \(error)")
-        return
-      }
-
-      let selectQuery = Select(from: titles)
-
-      connection.execute(query: selectQuery) { selectResult in
-        guard let resultSet = selectResult.asResultSet else {
-          guard let error = selectResult.asError else {
-            return callback("Error executing query: Unknown Error")
-          }
-          return callback("Error executing query: \(error)")
-        }
-        var resultString = ""
-
-      }
+      <p>Next we need iterate through our returned rows and convert them to books:</p>
+      <pre><code class="language-swift">var books = [Book]()
+resultSet.forEach() { row, error in
+    // Handle callback here
+}</code></pre>
+      <p>The `forEach` function will return either a `row`, an `error` or `nil` (Meaning there are no more rows).</p>
+      <p>Inside the `forEach` callback, we need to handle these three cases:</p>
+      <pre><code class="language-swift">guard let row = row else {
+    if let error = error {
+        Log.error("Error getting row: \(error)")
+        return completion(nil, .internalServerError)
+    } else {
+        // All rows have been processed
+        return completion(books, nil)
     }
-  }</code></pre>
-      <p>Since we have confirmed we have a valid result set we can get the column titles using the `getColumnTitles` method:</p>
-      <pre><code class="language-swift">func getTitles(_ callback: @escaping (String) -> Void ) -> Void {
-    pool.getConnection() { connection, error in
-      guard let connection = connection else {
-        guard let error = error else {
-          Log.error("Error connecting: Unknown Error")
-          return
-        }
-        Log.error("Error connecting: \(error)")
-        return
-      }
+}
+// Convert row to book here</code></pre>
+<p>If we get a row back from the database, We need to convert it to a `Book`.</p>
```

Comment:
"When we get a row back from the database we need to convert it back into a `Book` model type."